qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -206,284 +219,101 @@ def _sequences_to_circuits(
-        next_circ: QuantumCircuit,
-        composed_cliff_num: int,
-        qubits: List[int],
-    ):
-        """Append a Clifford to the end of a circuit. Return both the updated circuit and the updated
-        number representing the circuit"""
-        circ.compose(next_circ, inplace=True)
-        composed_cliff_num = StandardRB._clifford_utils.compose_num_with_clifford(
-            composed_num=composed_cliff_num,
-            qc=next_circ,
-        )
-        circ.barrier(qubits)
-        return circ, composed_cliff_num
-
-    def _add_inverse_to_circ(self, rb_circ, composed_num, qubits, clbits):
-        """Append the inverse of a circuit to the end of the circuit"""
-        inverse_cliff = StandardRB._clifford_utils.inverse_cliff(composed_num)
-        rb_circ.compose(inverse_cliff, inplace=True)
-        rb_circ.measure(qubits, clbits)
-        return rb_circ
-
-    # This method does a quick layout to avoid calling 'transpile()' which is
-    # very costly in performance
-    # We simply copy the circuit to a new circuit where we define the mapping
-    # of the qubit to the single physical qubit that was requested by the user
-    # This is a hack, and would be better if transpile() implemented it.
-    # Something similar is done in ParallelExperiment._combined_circuits
-    def _layout_for_rb(self):
-        transpiled = []
-        qargs_map = (
-            {0: self.physical_qubits[0]}
-            if self.num_qubits == 1
-            else {0: self.physical_qubits[0], 1: self.physical_qubits[1]}
-        )
-        for circ in self.circuits():
-            new_circ = QuantumCircuit(
-                *circ.qregs,
-                name=circ.name,
-                global_phase=circ.global_phase,
-                metadata=circ.metadata.copy(),
-            )
-            clbits = circ.num_clbits
-            if clbits:
-                creg = ClassicalRegister(clbits)
-                new_cargs = [Clbit(creg, i) for i in range(clbits)]
-                new_circ.add_register(creg)
-
-            for inst, qargs, cargs in circ.data:
-                mapped_cargs = [new_cargs[circ.find_bit(clbit).index] for clbit in cargs]
-                mapped_qargs = [circ.qubits[qargs_map[circ.find_bit(i).index]] for i in qargs]
-                new_circ.data.append((inst, mapped_qargs, mapped_cargs))
-                # Add the calibrations
-                for gate, cals in circ.calibrations.items():
-                    for key, sched in cals.items():
-                        new_circ.add_calibration(gate, qubits=key[0], schedule=sched, params=key[1])
-
-            transpiled.append(new_circ)
-        return transpiled
+        return op.adjoint()
 
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
-        if self.num_qubits in [1, 2]:
-            transpiled = self._layout_for_rb()
+        has_custom_transpile_option = (
+            any(
```

Comment:
Addressed in [cee4e78](https://github.com/Qiskit/qiskit-experiments/pull/922/commits/cee4e78b7f3e9faa6e51ab942893f3aaf3a418a1)

Reply:
I prefer to have warning or completely remove transpile options, i.e overriding `_set_transpile_options` method to raise warning/error. Usually users are not aware of what is happening inside `.run()` and may set some random option from random example code. I think having `inst_map` only makes sense in this experiment.

Reply:
Good suggestion. I like the idea restricting acceptable transpile options for each experiment. (too many possible `transpile_options` was one of my headaches.) For RB experiments, I don't like to but I'll accept `basis_gates` and `optimization_level` for now as well as `inst_map`. Because they are widely used in the tests (Fortunately, they seem not to be used in the tutorial). Forbidding them would require large refactoring of tests, so if we want that, I think it should be done in a separate PR.